Refactor/12 typography

JerryLiao-GoSky opened the pull request:
## 目的

closes #12 

## 實際工時

7.5 小時

- 2022/01/06 2 h
- 2022/01/05 5.5 h

## 如何測試

1. `yarn storybook`
2. 測試 `Text`、`Title`、`Paragraph` 各個子項目

## 預期結果

Props 皆參考 Ant design [Typography](https://ant.design/components/typography/) 的部分，除了 `<Title>` 有額外支援 `strong` 來加粗文字（antd `Title` 預設粗體，無法調粗細）
樣式應符合[設計圖](https://www.figma.com/file/og8KJ9f4UINpdYPb3slesI/GoSky-Library-(Lo-si)?node-id=715%3A658)

## 其它修改

- PR template 改年份
- 新增 `tuple.ts` 來處理神奇的 tuple type

JerryLiao-GoSky committed:
fix: default line-height

JerryLiao-GoSky committed:
fix: default line-height

JerryLiao-GoSky committed:
fix: modify colors

JerryLiao-GoSky committed:
fix: modify colors

JerryLiao-GoSky committed:
fix: style of text

JerryLiao-GoSky committed:
fix: style of text

JerryLiao-GoSky committed:
fix: style of Text

JerryLiao-GoSky committed:
fix: style of Text

JerryLiao-GoSky committed:
fix: template of Text for ellipsis

JerryLiao-GoSky committed:
fix: template of Text for ellipsis

JerryLiao-GoSky committed:
refactor: title component

JerryLiao-GoSky committed:
refactor: title component

JerryLiao-GoSky committed:
refactor: paragraph

JerryLiao-GoSky committed:
refactor: paragraph

JerryLiao-GoSky committed:
Merge branch 'refactor/12-typography' of github-jerryliao.com:goskyai/lo-si into refactor/12-typography

JerryLiao-GoSky committed:
fix: modify default year in PR template

JerryLiao-GoSky committed:
fix: lint warning

leo-gosky commented on src/assets/theme/antd/light-theme.less:
```diff
@@ -19,7 +19,7 @@
 @green-1: color(~`colorPalette('@{green-base}', 1) `);
 @green-2: #e9faeb;
-@green-3: #d3f5d6;
+@green-3: #aacbff;
```
它又變藍色惹

leo-gosky commented on src/components/Tabs/Tabs.stories.tsx:
```diff
@@ -40,13 +39,13 @@ const Template: Story = (args) => {
         <TabPanel
           tab={
             <Box display="flex" alignItems="center">
-              <Text mr="3">{tab.tab}</Text>
+              <Text>{tab.tab}</Text>
               <Icon icon={tab.icon} />
             </Box>
```
發現這邊 active 文字的顏色依然是套用 `.antd-typography` 的顏色。

看了 antd 的[範例](https://ant.design/components/tabs/#components-tabs-demo-icon)
要改一下寫法惹

JerryLiao-GoSky replied:
hmmmm 看起來 antd 的 icon 有 `margin-right` 是預設寫好的
之後改 Icon 再看看 antd 的 Custom Icon 會不會有一樣的效果
![image](https://user-images.githubusercontent.com/69893031/148357416-5486d262-77b0-4416-b908-2024ea72e641.png)
先刪掉會影響顏色 Text，之後 Tabs 的 PR 上會再完整的處理這塊

leo-gosky commented on src/components/Text/Text.stories.tsx:
```diff
@@ -1,18 +1,62 @@
 import { ComponentMeta, ComponentStory } from '@storybook/react';
-
+import faker from 'faker';
 import { Text } from '.';
```
Faker 確定掛惹，雖然用的是 `"faker": "^5.5.3"` 日後安裝也不會被影響到。
不過還是看看要不要把 `^` 拿掉，還是改用對外宣稱要接手的這個 [npm](https://www.npmjs.com/package/@withshepherd/faker)

JerryLiao-GoSky replied:
先改固定版本好了
然後大家感快 Fork 一份哦！

leo-gosky reviewed (changes requested):
不知道你有沒有遇到
```sh
[15:42:34] The following tasks did not complete: auto-update-theme, watch-themes-change
[15:42:34] Did you forget to signal async completion?
```
看要不要挑個[解法](https://stackoverflow.com/questions/36897877/gulp-error-the-following-tasks-did-not-complete-did-you-forget-to-signal-async)

JerryLiao-GoSky committed:
fix: color

JerryLiao-GoSky committed:
fix: remove Text in Tabs

JerryLiao-GoSky committed:
fix: typo

JerryLiao-GoSky committed:
fix: fixed the version of faker

JerryLiao-GoSky committed:
Merge branch 'refactor/10-redesign-components' into refactor/12-typography

JerryLiao-GoSky committed:
fix: order of starting storybook

[optional body]

[optional footer(s)]

# Commit 規範：
# - <type> 此 commit 的類型，常用如下
#   - feat  標示為新功能，對應 minor 版本號碼
#   - fix   問題修正，對應 patch 版本號碼
#   - test  補充測試、修正測試功能
#   - chore 其餘雜事，不影響任何功能
# - [scope] 選填項目，揭示此 commit 對應到的範圍
# - <description> 簡單扼要的描述 commit 的目的
# - [body] 詳細描述這段程式的主要目的
# - [footer] 可用來標記對應的 Ticket 或是 Issue 等等
#
# 範例:
# -----------------------------------------------------------
# feat(coupon): create coupon form
#
# 1. 新增建立 coupon 的表單
# 2. 新增 coupon fragment
# 3. 新增 CreateCouponQuery
#
# [Ticket: #123] [Issue: #123]
# -----------------------------------------------------------
# 注意 Subject 與 body、Footer 之間的空白不要刪掉惹
# 詳細參考 https://www.conventionalcommits.org/zh-hant/v1.0.0

JerryLiao-GoSky commented:
> 不知道你有沒有遇到
> 
> ```shell
> [15:42:34] The following tasks did not complete: auto-update-theme, watch-themes-change
> [15:42:34] Did you forget to signal async completion?
> ```
> 
> 看要不要挑個[解法](https://stackoverflow.com/questions/36897877/gulp-error-the-following-tasks-did-not-complete-did-you-forget-to-signal-async)

看起來 `themes-compile` 原本就有透過 `done()` 達成 async 的效果，因此後續的任務可以得知 `themes-compile` 什麼時候結束，
而 `watch-themes-change` 他是持續運作的，即使加了 Promise 也無法讓下一個任務得知 `watch-themes-change` 已經完成，
接續在 `watch-themes-change` 後面的是 `yarn storybook`，既然兩者都是持續運作的，
我想說就讓 `themes-compile` 先執行，執行完成後，讓 `watch-themes-change` 、`yarn storybook` 併發，
因為後面兩者大致上不會有衝突，不過 `themes-compile` 沒執行完的話，的確有可能影響 `yarn storybook`。

因此改了下 Script，刪掉了沒用到的 gulp tasks，哥再 Pull 下來看看ㄅ

leo-gosky reviewed (approved):
LGTM
